core/src/main/java/io/questdb/cairo/TableWriterAPI.java

```diff
@@ -76,6 +76,10 @@ public interface TableWriterAPI extends Closeable {
 
     void ic(long o3MaxLag);
 
+    default boolean isPartitionReadOnlyByTimestamp(long timestamp) {
```

Comment:
The TableWriter, on O3, detects that the partition is read only and aborts the transaction as a way of punishment. The rationale is that only partitions attached via soft link are read only, which means you set them that way, which means you know what you are doing, so you should avoid inserts/updates on such partitions. 

I have changed the default so that an exception is thrown.

Reply:
for the Line protocol, I silently ignore rows that fall into a read only partition, because I have the row timestamp and I can check whether it belons to a read only partition.

Reply:
Can we instead silently ignore the lines in `TableWriter` on applying O3 data?
This is the only way I see it can work for wal table.

Reply:
yes, this is the way it is now